packages/terra-form-checkbox/src/Checkbox.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './Checkbox.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+    * The checked property of the Input element. Use this to generate a controlled Checkbox Element.
+    */
+  checked: PropTypes.bool,
+  /**
+    * The checked property of the Input element. Use this to generate an uncontrolled Checkbox Element.
+    */
+  defaultChecked: PropTypes.bool,
+  /**
+    * The id of the input field.
+    */
+  id: PropTypes.string,
+  /**
+    * Additional attributes for the input object.
+    */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the checkbox element is disabled or not.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+    * Whether the checkbox element is inline or not.
+    */
+  isInline: PropTypes.bool,
+  /**
+    * Whether the label is hidden or not.
+    */
+  isLabelHidden: PropTypes.bool,
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+  /**
+    * Additional objects for the text object.
+    */
+  labelTextAttrs: PropTypes.object,
+  /**
+    * Name of the input attribute.
+    */
+  name: PropTypes.string,
+  /**
+    * Function to trigger when user clicks on the input. Provide a function to create a controlled input.
+    */
+  onChange: PropTypes.func,
+  /**
+    * The value of the input element.
+    */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  checked: undefined,
```

Comment:
Since checked and defaultChecked are booleans, these should be false, not undefined

Reply:
I believe that we need these set to undefined to start with so that we can distinguish between a controlled and an uncontrolled checkbox. defaultChecked maybe ok to set to false, but we will need to make sure checked has an actual value passed in to verify the component should be controlled.

Reply:
The logic for this is based on discussion within the old control component, [Control.jsx](https://github.com/cerner/terra-core/pull/448/files), that you were a part of hehe.